app/containers/Products/ProductRowItemContainer.tsx

```diff
@@ -36,430 +80,369 @@ export const ProductRowItemComponent = props => {
+                'DD-MM-YYYYTHH:mm:ss'
+              ).format('YYYY-MM-DDTHH:mm:ss')
+            : null
+        }
       }
-    });
-
-    if (validBenchmarks.length > 1) {
-      console.error(
-        'Too many benchmarks already created, only one active benchmark and one upcoming benchmark can be defined at one time'
-      );
-      return;
-    }
+    };
+    const response = await editBenchmarkMutation(relay.environment, variables);
+    console.log(response);
+  };
 
-    const benchmarkTargetIndex = 3;
-    const eligibilityThresholdTargetIndex = 4;
-    const newVariables = {
+  // This fuction does not 'DELETE' from the database, but sets the deleted at / deleted by values. This action is not recoverable through the UI
+  const deleteBenchmark = async benchmark => {
+    if (!benchmark) return;
+    const variables = {
       input: {
-        productIdInput: props.product.rowId,
-        benchmarkInput: parseFloat(
-          event.nativeEvent.target[benchmarkTargetIndex].value
-        ),
-        eligibilityThresholdInput: parseFloat(
-          event.nativeEvent.target[eligibilityThresholdTargetIndex].value
-        ),
-        startDateInput: startDate,
-        prevBenchmarkIdInput: currentBenchmark ? currentBenchmark.rowId : null
+        id: currentBenchmark.id,
+        benchmarkPatch: {
+          deletedAt: moment(),
+          deletedBy: 'Admin'
+        }
       }
     };
-    const response = await createBenchmarkMutation(
-      props.relay.environment,
-      newVariables
-    );
+    const response = await editBenchmarkMutation(relay.environment, variables);
     console.log(response);
   };
 
-  /** Mutations & functions above */
-  /** Code for Rendering Below */
-  const {product} = props;
-  // Get the current benchmark for the product
-  let benchmarks;
-  if (props.product.benchmarksByProductId.edges[0]) {
-    props.product.benchmarksByProductId.edges.forEach(({node: benchmark}) => {
-      if (
-        Date.parse(benchmark.startDate) < Date.now() &&
-        (benchmark.endDate === null ||
-          Date.parse(benchmark.endDate) > Date.now()) &&
-        !benchmark.deletedAt
-      ) {
-        benchmarks = benchmark;
-      }
-    });
-    if (!benchmarks) {
-      benchmarks = {benchmark: '', eligibilityThreshold: ''};
-    }
-  } else {
-    benchmarks = {benchmark: '', eligibilityThreshold: ''};
-  }
+  const productFormData = {
+    product: product.name,
+    description: product.description
+  };
 
-  // Archived logic to determine display values
-  const background = props.product.state === 'archived' ? 'lightGrey' : '';
-  const buttonVariant =
-    props.product.state === 'archived' ? 'success' : 'warning';
-  const archiveRestore =
-    props.product.state === 'archived' ? 'Restore' : 'Archive';
+  const currentBenchmarkFormData = {
+    benchmark:
+      currentBenchmark && currentBenchmark.benchmark
+        ? currentBenchmark.benchmark
+        : null,
```

Comment:
This can now be written this way: `benchmark: currentBenchmark?.benchmark ?? null`